google-cloud-firestore/lib/google/cloud/firestore/document_reference.rb

```diff
@@ -0,0 +1,387 @@
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a document reference
+        #   nyc_ref = firestore.doc "cities/NYC"
+        #
+        #   nyc_ref.create({ name: "New York City" })
+        #
+        def create data
+          ensure_client!
+
+          client.batch { |b| b.create self, data }
+        end
+
+        ##
+        # Write the provided data (fields and values) to the document. If the
+        # document does not exist, it will be created. By default, the provided
+        # data overwrites existing data, but the provided data can be merged
+        # into the existing document using the `merge` argument.
+        #
+        # If you're not sure whether the document exists, use the `merge`
+        # argument to merge the new data with any existing document data to
+        # avoid overwriting entire documents.
+        #
+        # @param [Hash] data The document's fields and values.
+        # @param [Boolean, String, Symbol, Array<String|Symbol>] merge When
+        #   `true`, all provided data is merged with the existing document data.
+        #   When the argument is a field path or paths, only the data for fields
+        #   in this argument is merged with the existing document data. The
+        #   default is to not merge, but to instead overwrite the existing
+        #   document data.
+        #
+        # @return [CommitResponse] The response from committing the changes.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a document reference
+        #   nyc_ref = firestore.doc "cities/NYC"
+        #
+        #   nyc_ref.set({ name: "New York City" })
+        #
+        # @example Set a document and merge all data:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a document reference
+        #   nyc_ref = firestore.doc "cities/NYC"
+        #
+        #   nyc_ref.set({ name: "New York City" }, merge: true)
+        #
+        # @example Set a document and merge only name:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a document reference
+        #   nyc_ref = firestore.doc "cities/NYC"
+        #
+        #   nyc_ref.set({ name: "New York City" }, merge: [:name])
+        #
+        def set data, merge: nil
+          ensure_client!
+
+          client.batch { |b| b.set self, data, merge: merge }
+        end
+
+        ##
+        # Update the document with the provided data (fields and values). The
+        # provided data is merged into the existing document data.
+        #
+        # The operation will fail if the document does not exist.
```

Comment:
As stated above, please state that update accepts field paths.

Reply:
The parameter documentation has been updated.